src/squashdelta.cxx

```diff
@@ -33,16 +33,16 @@ extern "C"
 
 struct compressed_block
 {
-	size_t offset;
-	size_t length;
-	size_t uncompressed_length;
+	uint64_t offset;
+	uint32_t length;
+	uint32_t uncompressed_length;
```

Comment:
Does this change make it more correct, if we were to process a delta on a 32-bit system for some reason (who knows if we try to optimize robot costs)? Or would the rest of the program not work anyway on such a system?

Reply:
I'm not 100% sure I understand how size_t gets set. We are already compiling on 64bit systems.

Reply:
That's a tricky one, but I would say this is more "correct".
While size of `size_t` changes based on system/compiler (usually 32 bits in a 32-bits system,  and 64 in a 64-bits system), compilers nowadays handle uint64_t in 32-bits environments. While performance is slower, it allows you to represent a 64 bit data (which would still be necessary to process squashfs images > 4GB).

Reply:
After thinking some more, I don't think this program would work with large files on a 32-bit system, because it reads the source squashfs file via a memory map.

Reply:
To clarify Tony, I added this change right now and am asking if it's correct. (Because you can't comment on a line on GitHub without changing it.)

Reply:
Now I'm thinking `size_t` is more correct, because these are representing offsets and lengths in the memory map.